game-core/src/main/java/games/strategy/engine/lobby/client/login/LobbyServerPropertiesFetcher.java

```diff
@@ -25,6 +27,8 @@
 @Log
 public final class LobbyServerPropertiesFetcher {
   private final Function<String, Optional<File>> fileDownloader;
+  @Nullable private LobbyServerProperties lobbyServerProperties;
```

Comment:
For future reference, there is Guava's `Suppliers#memoize()` to take care of all this.  There's also a variant to expire the cached value after a period of time.  For example:

```diff
diff --git a/game-core/src/main/java/games/strategy/engine/lobby/client/login/LobbyServerPropertiesFetcher.java b/game-core/src/main/java/games/strategy/engine/lobby/client/login/LobbyServerPropertiesFetcher.java
index e5984229d..11d68880d 100644
--- a/game-core/src/main/java/games/strategy/engine/lobby/client/login/LobbyServerPropertiesFetcher.java
+++ b/game-core/src/main/java/games/strategy/engine/lobby/client/login/LobbyServerPropertiesFetcher.java
@@ -5,13 +5,14 @@ import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.util.Optional;
+import java.util.concurrent.TimeUnit;
 import java.util.function.BiFunction;
 import java.util.function.Function;
+import java.util.function.Supplier;
 import java.util.logging.Level;
 
-import javax.annotation.Nullable;
-
 import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Suppliers;
 
 import games.strategy.engine.ClientContext;
 import games.strategy.engine.framework.map.download.DownloadConfiguration;
@@ -27,8 +28,8 @@ import lombok.extern.java.Log;
 @Log
 public final class LobbyServerPropertiesFetcher {
   private final Function<String, Optional<File>> fileDownloader;
-  @Nullable private LobbyServerProperties lobbyServerProperties;
-
+  private final Supplier<Optional<LobbyServerProperties>> fetchLobbyServerProperties =
+      Suppliers.memoizeWithExpiration(this::fetchProperties, 1L, TimeUnit.HOURS);
 
   /**
    * Default constructor with default (prod) dependencies.
@@ -60,13 +61,9 @@ public final class LobbyServerPropertiesFetcher {
    *         Otherwise backup values from client config.
    */
   public Optional<LobbyServerProperties> fetchLobbyServerProperties() {
-    if (lobbyServerProperties == null) {
-      final Optional<LobbyServerProperties> props = fetchProperties();
-      props.ifPresent(lobbyProps -> lobbyServerProperties = lobbyProps);
-    }
-    return Optional.ofNullable(lobbyServerProperties);
+    return fetchLobbyServerProperties.get();
   }
-  
+
   private Optional<LobbyServerProperties> fetchProperties() {
     final Optional<LobbyServerProperties> userOverride = getTestOverrideProperties();
     if (userOverride.isPresent()) {
```